src/eos-autoupdater.c

```diff
@@ -421,86 +512,145 @@ is_connected_through_mobile (void)
   NMDevice *device;
   const GPtrArray *devices;
   gboolean is_mobile = FALSE;
-  gint i;
+  guint i;
 
   client = nm_client_new ();
   if (!client) {
     return FALSE;
   }
 
-  g_object_get (client, "primary-connection", &connection, NULL);
+  connection = nm_client_get_primary_connection (client);
   if (!connection) {
     g_object_unref (client);
     return FALSE;
   }
 
   devices = nm_active_connection_get_devices (connection);
-  for (i = 0; i < devices->len; i++) {
+  for (i = 0; !is_mobile && i < devices->len; i++) {
     device = (NMDevice *) g_ptr_array_index (devices, i);
     switch (nm_device_get_device_type (device)) {
     case NM_DEVICE_TYPE_MODEM:
     case NM_DEVICE_TYPE_BT:
     case NM_DEVICE_TYPE_WIMAX:
-      is_mobile |= TRUE;
+      is_mobile = TRUE;
       break;
+    case NM_DEVICE_TYPE_UNKNOWN:
+    case NM_DEVICE_TYPE_ETHERNET:
+    case NM_DEVICE_TYPE_WIFI:
+    case NM_DEVICE_TYPE_UNUSED1:
+    case NM_DEVICE_TYPE_UNUSED2:
+    case NM_DEVICE_TYPE_OLPC_MESH:
+    case NM_DEVICE_TYPE_INFINIBAND:
+    case NM_DEVICE_TYPE_BOND:
+    case NM_DEVICE_TYPE_VLAN:
+    case NM_DEVICE_TYPE_ADSL:
+    case NM_DEVICE_TYPE_BRIDGE:
+    case NM_DEVICE_TYPE_GENERIC:
+    case NM_DEVICE_TYPE_TEAM:
+    case NM_DEVICE_TYPE_TUN:
+    case NM_DEVICE_TYPE_IP_TUNNEL:
+    case NM_DEVICE_TYPE_MACVLAN:
+    case NM_DEVICE_TYPE_VXLAN:
+    case NM_DEVICE_TYPE_VETH:
     default:
       break;
     }
   }
 
-  g_object_unref (connection);
   g_object_unref (client);
 
   return is_mobile;
 }
 
+static gboolean
+should_listen_on_session_bus (void)
+{
+  const gchar *value = NULL;
+
+  value = g_getenv ("EOS_UPDATER_TEST_AUTOUPDATER_USE_SESSION_BUS");
+
+  return value != NULL;
+}
+
+static gint
+get_dbus_timeout (void)
+{
+  const gchar *value = NULL;
+  gint64 timeout;
+  gchar *str_end = NULL;
+
+  value = get_envvar_or ("EOS_UPDATER_TEST_AUTOUPDATER_DBUS_TIMEOUT",
+                         NULL);
+
+  if (value == NULL || value[0] == '\0')
+    return -1;
+
+  errno = 0;
+  timeout = g_ascii_strtoll (value, &str_end, 10);
+  if (errno != 0 ||
+      str_end == NULL ||
+      *str_end != '\0' ||
+      timeout > G_MAXINT ||
+      timeout < 0)
+    return -1;
+
+  return timeout;
+}
+
 int
 main (int argc, char **argv)
 {
   EosUpdater *proxy;
   GError *error = NULL;
-  gint update_interval;
+  guint update_interval_days;
   gboolean update_on_mobile;
   gboolean force_update = FALSE;
   GOptionContext *context;
 
   GOptionEntry entries[] = {
     { "force-update", 0, 0, G_OPTION_ARG_NONE, &force_update, "Force an update", NULL },
+    { "from-volume", 0, 0, G_OPTION_ARG_STRING, &volume_path, "Poll for updates from the volume", "PATH" },
```

Comment:
It’s from commit 397b3040:
> Allow autoupdater to run updates from volume
>
> This adds the --from-volume flag that takes a string. Using this flag
> makes autoupdater to call PollVolume instead of Poll.

I have no idea of the motivation for this. Maybe to allow machines to be configured to always poll a mounted network volume (on a local server) for updates, rather than the network?